### PR TITLE
Template for release body

### LINKED ISSUE #9: Let the user specify a per-commit template
Instead of using `include-hash`, this could allow a per-commit template so the user can customize it freely

```yml
with:
  template: '- {hash} {title}' # Default
```

```yml
with:
  template: '- {title} {hash}'
```
```yml
with:
  template: '{title} [🔗]({url})'
```

### PR DESCRIPTION
**Pull request by notlmn**
Closes #9.

Adds new options for specifying:
1. an overall release notes template (`template`)
	with fields ~`{version}`~, `{commits}`, and `{range}`
2. commit entry template (`commit-template`)
	with fields `{hash}`, `{title}`, and `{url}`

Also went ahead and updated the readme a bit, for documenting new options, and updating the screenshot.

### CODE REVIEW

File: action.yml [hunk 1]
```diff
@@ -11,26 +11,23 @@ inputs:
...
-  header:
-    description: Content to prepend at the start of release notes
+  template:
+    description: Markdown template to be included in release notes
     required: false
-    default: ''
-  footer:
-    description: Content to append at the end of release notes
+    default: |
+      ### Changelog
```
Comment (fregante) [inline action.yml]: This is not the current default, I'd drop it

File: action.yml [hunk 2]
```diff
@@ -11,26 +11,23 @@ inputs:
...
+
+      {commits}
+
+      {range}
+  commit-template:
+    description: Template for each commit entry in release notes
```
Comment (fregante) [inline action.yml]:
```suggestion
    description: Markdown template for each commit entry in release notes
```

File: index.js [hunk 3]
```diff
@@ -57,34 +54,28 @@ async function run() {
...
-		}
-
 		const octokit = getOctokit(core.getInput('token'));
 		const createReleaseResponse = await octokit.repos.createRelease({
 			repo,
 			owner,
 			tag_name: pushedTag, // eslint-disable-line camelcase
-			body: releaseBody.join('\n'),
+			body: releaseTemplate
+				.replace('{version}', pushedTag)
```
Comment (fregante) [inline index.js]: This is already the title of the tag, I don't think it's needed.

File: readme.md [hunk 4]
```diff
@@ -31,33 +33,28 @@ The action expects you to do a deep clone of the repository using `actions/check
...
 
-### include-hash
+{commits}
 
-Default: `false`
+{range}
+```
 
-Prepend and link commit hash to each entry.
+Markdown template to be included in release notes.
```
Comment (fregante) [inline readme.md]:
```suggestion
Markdown template to be included in release notes. Available replacements:

- `{commits}` description here
- `{range}` description here
- etc
```

File: index.js (hunk 3 above)
Comment (notlmn) [reply index.js]: It is only replaced if it exists, probably for people who want to generate `changelog.md` from releases.

Also thinking about it now, feels dumb. Will remove this.

File: readme.md [hunk 5]
```diff
@@ -31,33 +33,32 @@ The action expects you to do a deep clone of the repository using `actions/check
...
 
 Required: [Personal access token](https://docs.github.com/en/github/authenticating-to-github/creating-a-personal-access-token) used to create releases.
 
-### footer
+### template
 
-Default: `''`
+Default:
+``` yml
+### Changelog
```
Comment (fregante) [inline readme.md]: Also here

File: readme.md [hunk 6]
```diff
@@ -31,33 +33,32 @@ The action expects you to do a deep clone of the repository using `actions/check
...
+{range}
+```
 
-Default: `false`
+Markdown template to be included in release notes. Available replacements:
 
-Prepend and link commit hash to each entry.
+- `{commits}` description here
+- `{range}` description here
+- etc
```
Comment (fregante) [inline readme.md]: 🤔

**Comment by fregante** [comment]
LGTM!

**Comment by notlmn** [comment]
Thanks for the review @fregante!

Leaving the releases out to you, in case you want to get #14 merged before releasing `v3`

**Pull request merged**